src/engine.rs

```diff
@@ -146,15 +175,15 @@ mod tests {
 
     #[test]
     fn reqwest_engine_can_collect_facts() {
-        let eng = Engine::new(vec!["https://www.google.com".to_string()]);
+        let eng = Engine::new(vec!["https://www.google.com".to_string()], vec![]);
         let mut fact: Option<Fact> = None;
         eng.run(1, |f| fact = Some(f));
         assert!(fact.is_some());
     }
 
     #[test]
     fn hyper_engine_can_collect_facts() {
-        let eng = Engine::new(vec!["https://www.google.com".to_string()]).with_hyper();
+        let eng = Engine::new(vec!["https://www.google.com".to_string()], vec![]).with_hyper();
```

Comment:
I wonder if there's a way to devise a test that ensures that the headers are passed forward?

Reply:
Good idea. We can use httpbin.org for this.

Reply:
@kbacha I'll have some time to finish this up today. Since, httpbin returns the headers in the response body, we can do one of three things:
1. Abstract away the actual request functionality from both hyper and reqwest so that we have access to the response.
2. Start returning the response in the collector as well.
3. Make two requests (with and without headers), and then use the content length to measure the difference. Obviously, a request with headers will be bigger.

What do you propose?

Reply:
Ok, let me think and reflect:

1. This could work and I've been toying with this approach on my own but haven't built anything to do this yet. Some sort of trait seems to make sense. If you feel like taking that on I think it could benefit the tool in general, but is a much large lift.
2. I suppose we could add the response to the `Fact` but the body is purposes discarded after the length and fact is recorded. Could make this an parameter to include it but I don't generally like a parameter that only makes sense for testing.
3. I like this as a simple approach. The byte size difference should be calculable based on the size of the header key and value, correct? In which case, it's a pretty solid approach.

I'd say that option 3 is a great approach for this PR. If you wish to dive into a better abstraction, I'm supportive of that too.